// Copyright (C) 2022 The Android Open Source Project
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package ninja_metrics;

// NinjaMetrics is to deliver the information of input files to build a target
// device or an image. These information is captured from ninja analysis
message NinjaMetrics {
    // Total number of input files to build a target or an image
    uint32 num_input_files = 1;

    // The number of git projects that need to be checked out to build a target
    // or an image
    uint32 num_projects = 2;

    // Details of the all required git projects
    repeated ProjectInfo project_infos = 3;
}

// ProjectInfo is to describe each git project
message ProjectInfo {
    // Path of the git project
    string name = 1;

    // The number of required files in the git project to build a target or an
    // image
    uint32 num_input_files = 2;
}
